From e5fa6d10b755e38eb11eb7d1bd36277b852099a4 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Thu, 12 Sep 2019 13:05:37 +0200
Subject: [PATCH 03/12] [first stage] If Vboot2 fails, fall-back to Vboot1

Some devices, for instance Honor View 10, running Pie vendor declares
vbmeta in their device-tree, but doesn't have a vbmeta partition.

Test: without this fix the device reboots into bootloader
Test: with this fix, the device boots.

Please note that other fixes are also required to get this device to
actually boot.

Change-Id: I97a7042fb03f817c41b801d558e438d2c1f6c375
---
 init/first_stage_mount.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/init/first_stage_mount.cpp b/init/first_stage_mount.cpp
index 8eb2f9748..990d76928 100644
--- a/init/first_stage_mount.cpp
+++ b/init/first_stage_mount.cpp
@@ -809,7 +809,13 @@ bool DoFirstStageMount() {
         LOG(ERROR) << "Failed to create FirstStageMount";
         return false;
     }
-    return handle->DoFirstStageMount();
+    if(!handle->DoFirstStageMount()) {
+	    handle = nullptr;
+	    auto fstab = ReadFirstStageFstab();
+	    std::unique_ptr<FirstStageMount> v = std::make_unique<FirstStageMountVBootV1>(std::move(fstab));
+	    return v->DoFirstStageMount();
+    }
+    return true;
 }
 
 void SetInitAvbVersionInRecovery() {
-- 
2.25.1

